pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -345,39 +359,87 @@ func (r *ReconcileCnsFileAccessConfig) Reconcile(request reconcile.Request) (rec
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileCnsFileAccessConfig) configureFileVolumeACL(ctx context.Context, volumeID string, vm *vmoperatortypes.VirtualMachine, instance *cnsfileaccessconfigv1alpha1.CnsFileAccessConfig) error {
+// configureFileVolumeACLWithDelete helps to remove net permissions for file volumes
+// Net permissions will be removed when the external facing IP is the only entry in
+// the ExternalIPtoClientVms list obtained from cnsfilevolumeclient instance
+// For other cases, the method will remove the vm entry from ExternalIPtoClientVms
+//  Returns error if any operation fails
+func (r *ReconcileCnsFileAccessConfig) configureFileVolumeACLWithDelete(ctx context.Context, volumeID string, vm *vmoperatortypes.VirtualMachine, instance *cnsfileaccessconfigv1alpha1.CnsFileAccessConfig) error {
```

Comment:
Rename to something more relatable, may be `removePermissionsFromFileVolumeACL`? Similar for the other method to add permissions.

Reply:
Why can't this be refactored with `configureFileVolumeACL`? There's quite a bit of common code, just a slight change in logic (removing/adding ip addresses)

Reply:
Done, refactored to accommodate both the functionalities